cmd/dev.go

```diff
@@ -0,0 +1,160 @@
+// Architect will not build for windows, therefore we exclude the whole command for windows
+//go:build !windows
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/skatteetaten/ao/pkg/auroraconfig"
+	"github.com/skatteetaten/ao/pkg/deploymentspec"
+	"github.com/skatteetaten/ao/pkg/prompt"
+	architect "github.com/skatteetaten/architect/v2/pkg/build"
+	architectConfig "github.com/skatteetaten/architect/v2/pkg/config"
+	"github.com/spf13/cobra"
+	"os"
+	"strings"
+)
+
+const OpenshiftRegistryFormat = "default-route-openshift-image-registry.apps.%s.paas.skead.no"
+
+var (
+	devCmd = &cobra.Command{
+		Use:         "dev",
+		Short:       "Perform dev operations",
+		Annotations: map[string]string{"type": "development"},
+	}
+
+	rolloutCmd = &cobra.Command{
+		Use:   "rollout <applicationdeploymentRef> <Leveransepakke file>",
+		Short: "Rollout leveransepakke to openshift for development type",
+		RunE:  Rollout,
+	}
+)
+
+func init() {
+	devCmd.AddCommand(rolloutCmd)
+	RootCmd.AddCommand(devCmd)
+}
+
+func Rollout(cmd *cobra.Command, args []string) error {
+	if len(args) != 2 {
+		return cmd.Usage()
+	}
+
+	leveransePakkeFile := args[1]
+	if err := verifyFileExists(leveransePakkeFile); err != nil {
+		return err
+	}
+
+	fileNames, err := DefaultAPIClient.GetFileNames()
+	if err != nil {
+		return err
+	}
+
+	search := args[0]
+
+	matchedAdr, err := getFuzzyMatchedAdr(search, fileNames)
+	if err != nil {
+		return err
+	}
+
+	spec, err := getSpec(*matchedAdr)
+	if err != nil {
+		return err
+	}
+
+	deployType := spec.Get("type")
+	if deployType != "development" {
+		return fmt.Errorf("You need to specify type=development in AuroraConfig. Current type=%s", deployType)
+	}
+
+	conf := getArchitectBuildConfiguration(spec, leveransePakkeFile)
+
+	cmd.Println("Building image.. If you wish to see the details you can run the command with `-l info` flag.")
+	architect.BuildBinary(conf)
+
+	return nil
+}
+
+func getSpec(matchedAdr string) (deploymentspec.DeploymentSpec, error) {
+	allSpecs, err := DefaultAPIClient.GetAuroraDeploySpec([]string{matchedAdr}, !flagNoDefaults, flagIgnoreErrors)
```

Comment:
Bruken av flagg er litt ustandard.  Det er ingen tilordning av flagg til rolloutCmd i init() og da er det vanskelig å se at disse flaggene (flagNoDefaults og flagIgnoreErrors) faktisk vil ha noen annen verdi enn default fra der de er definert i denne konteksten.

Jeg ville også vurdert om det skal være mulig å skippe prompt ved å spesifisere disse flaggene tilsvarende som i deploy:
`	rolloutCmd.Flags().BoolVarP(&flagAcceptRollout, "yes", "y", false, "Suppress prompts and accept rollout")`
`	rolloutCmd.Flags().BoolVarP(&flagAcceptRollout, "no-prompt", "", false, "Suppress prompts and accept rollout")`
og så gjøre spørsmålet om rollout avhenging av flagget.